backend/src/cache/storage/execution_cache_store.go

```diff
@@ -76,7 +77,7 @@ func (s *ExecutionCacheStore) scanRows(rows *sql.Rows, podMaxCacheStaleness int6
 		}
 		log.Println("Get id: " + strconv.FormatInt(id, 10))
 		log.Println("Get template: " + executionTemplate)
-		if maxCacheStaleness == -1 || s.time.Now().UTC().Unix()-startedAtInSec <= podMaxCacheStaleness {
+		if maxCacheStaleness == -1 || s.time.Now().UTC().Unix()-startedAtInSec <= maxCacheStaleness {
```

Comment:
I think the check needs to be 
```suggestion
		if podMaxCacheStaleness == -1 || s.time.Now().UTC().Unix()-startedAtInSec <= podMaxCacheStaleness {
```
(with `podMaxCacheStaleness` parameter added back to the function).
The task cannot/should not control whether its data can be reused in the future. It can only control whether it can be skipped.
So, when we do lookup, we only care about the preferences of the current pod/task.